librarian_packages/hera_librarian/hera_librarian/base_store.py

```diff
@@ -79,9 +79,9 @@ def _ssh_slurp(self, command, input=None):
 
         if proc.returncode != 0:
             raise RPCError(argv, 'exit code %d; stdout:\n\n%r\n\nstderr:\n\n%r'
-                           % (proc.returncode, stdout, stderr))
+                           % (proc.returncode, stdout.decode("utf-8"), stderr.decode("utf-8")))
 
-        return stdout
+        return stdout.decode("utf-8")
```

Comment:
Just as a heads-up ... this is potentially dangerous (in the sense of this function throwing unexpected exceptions) in case anyone ever wants to use `_ssh_slurp` to dump data from a binary file, since that won't be generically decodable as UTF8. And uncooperative shell commands can find all sorts of reasons to emit output that isn't UTF8.

Reply:
Didn't realize this could be a problem, but it totally makes sense. I removed this decode statement and ended up putting ones in the tests, which is where I first realized the output of `_ssh_slurp` was no longer producing strings.